racket/collects/racket/init.rkt

```diff
@@ -12,3 +12,7 @@
 (provide (all-from-out racket
                        racket/enter
                        racket/help))
+
+;; Set the default replt to XREPL
+(with-handlers ([exn:fail? void])
```

Comment:
The major difference is that your version will swallow other errors, such as not having readline or editline available.